poke_redirect.py

```diff
@@ -0,0 +1,277 @@
+    size.to_bytes(1, byteorder=CODE_BYTEORDER) +
+        b'\x90\xa0\xe3',  # mov   r9, <size>       ; set filePath.size to the size of savePath
+    b'\x03\xa0\xa0\xe3',  # mov   r10, #0x3        ; set openFlags to OPEN_READ|OPEN_WRITE
+    b'\x00\xb0\xa0\xe3',  # mov   r11, #0x0        ; set attributes to 0
+    b'\xf0\x0f\x2d\xe9',  # stmdb sp!, { r4-r11 }  ; push variables
+    ASM_Branch(True, addr + 52, config.fs_open),   # bl FSUSER_OpenFileDirectly
+    b'\xf0\x0f\xbd\xe8',  # ldmia sp!, { r4-r11 }  ; pop variables
+    b'\xfc\x8f\xbd\xe8',  # ldmia sp!, { r2-r11, pc }
+
+    # fsHandle: .word <config.fs_handle>
+    config.fs_handle.to_bytes(4, byteorder=CODE_BYTEORDER),
+
+    # emptyString: .string ""
+    PadASMString(b''),
+
+    # savePath: .string <path_bytes>
+    PadASMString(path_bytes),
+  ))
+
+
+def CreatePatchRecords(config: Config, card_id: str, save_path: str) -> Dict[int, bytes]:
+  records = {}
+  sd_asm = b''
+  sd_addr = CODE_OFFSET + config.code_size + SD_FUNCTION_OFFSET
+
+  # Add OpenSDSaveFile function
+  open_addr = CODE_OFFSET + config.code_size + OPEN_FUNCTION_OFFSET
+  records[open_addr] = ASM_OpenSDSaveFile(config, open_addr, save_path)
+
+  # Redirect save read
+  read_addr = sd_addr + len(sd_asm)
+  sd_asm += ASM_ReadSDSaveFile(config, read_addr, open_addr)
+  records[config.read_save] = ASM_Branch(False, config.read_save, read_addr)
+
+  # Redirect save write
+  write_addr = sd_addr + len(sd_asm)
+  sd_asm += ASM_WriteSDSaveFile(config, write_addr, open_addr)
+  records[config.write_save] = ASM_Branch(False, config.write_save, write_addr)
+
+  # Redirect card check if needed
+  if config.check_card is not None:
+    check_addr = sd_addr + len(sd_asm)
+    sd_asm += ASM_CheckSDSaveFile(config, check_addr, open_addr)
+    records[config.check_card] = ASM_Branch(False, config.check_card, check_addr)
+  
+  # Add SD Functions
+  records[sd_addr] = sd_asm
+  
+  # Patch the function that reads the card id
+  skip_addr = config.card_id + CARD_ID_SKIP_OFFSET
+  fake_addr = config.card_id + CARD_ID_FAKE_OFFSET
+  records[skip_addr] = ASM_Branch(False, skip_addr, fake_addr)
+  records[fake_addr] = ASM_FakeCardId(card_id)
+
+  return records
+
+
+def CreatePatch(config: Config, card_id: str, save_path: str) -> bytes:
+  patch = [PATCH_HEADER]
+  for offset, data in sorted(
+      CreatePatchRecords(config, card_id, save_path).items(),
+      key=operator.itemgetter(0)):
+    patch.append((offset - CODE_OFFSET).to_bytes(3, byteorder=PATCH_BYTEORDER))
+    patch.append(len(data).to_bytes(2, byteorder=PATCH_BYTEORDER))
+    patch.append(data)
+  patch.append(PATCH_EOF)
+  return b''.join(patch)
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser(
+      description='Create an IPS patch to redirect card access to a file on SD')
+  parser.add_argument(
+      'config', choices=CONFIGS.keys(), help='which binary configuration to use')
+  parser.add_argument('card_id', help='the 4 character id of the card')
```

Comment:
What are thoughts on providing a default card Id and save path just in case users don't know the available card Ids or whether they need to prepend anything to the save path (i.e. `sd:/` or `/`)?  Defaults can serve as documentation :slightly_smiling_face:

An alternative would also be to provide an example of each in the description.

Reply:
I'd rather provide documentation than defaults.  I think defaults will lead to users just running the script without looking at it, and not realizing what the arguments are

Reply:
Sounds good to me